src/transformers/modeling_utils.py

```diff
@@ -550,8 +550,12 @@ def _load_state_dict_into_meta_model(
             param_name = param_name[len(start_prefix) :]
 
         module_name = param_name
-        # We convert floating dtypes to the `dtype` passed.
-        if dtype is not None and not str(param.dtype).startswith("torch.int"):
+
+        # We convert floating dtypes to the `dtype` passed. We make sure that the
+        # parameter is a floating point parameter. Sometimes we want to keep the buffers
+        # in int/uint/bool and not cast them.
+        # Please see: https://github.com/huggingface/transformers/pull/18471 for more details
```

Comment:
A bit long, let's leave it at:
```suggestion
        # We convert floating dtypes to the `dtype` passed.We want to keep the buffers/params
        # in int/uint/bool and not cast them.
```